solargeo/sunae.pyx

```diff
@@ -0,0 +1,247 @@
+            Dr. Lee Harrison (lee@asrc.albany.edu)
+            Atmospheri#Sciences Research Center
+            State University of New York
+            Albany, New York
+    (fortran) Modified by:  Dr. Warren Wiscombe (wiscombe@climate.gsfc.nasa.gov)
+                NASA Goddard Space Flight Center
+                Code 913
+                Greenbelt, MD 20771
+    (python)  Converted to python: Karl Lapo (lapo.karl@gmail.com)
+
+    See Also
+    --------
+    avg_el: average elevation angle
+     '''
+    #  Local Variables:
+    #    dec       Declination (radians)
+    #    eclon    Eclipti#longitude (radians)
+    #    gmst      Greenwich mean sidereal time (hours)
+    #    ha        Hour angle (radians, -pi to pi)
+    #    jd        Modified Julian date (number of days, including
+    #              fractions thereof, from Julian year J2000.);
+    #              actual Julian date is jd + 2451545.
+    #    lmst      Local mean sidereal time (radians)
+    #    mnanom    Mean anomaly (radians, normalized to 0 to 2*pi)
+    #    mnlon    Mean longitude of Sun, corrected for aberration
+    #              (deg; normalized to 0-360)
+    #    oblqec    Obliquity of the eclipti#(radians)
+    #    ra        Right ascension  (radians)
+    #    refrac    Refraction correction for US Standard Atmosphere (degs)
+
+    # Error handling
+    if np.min(year) < 1950 or np.max(year) > 2050:
+        raise ValueError('year must be between 1950 and 2050')
+    if np.min(day) < 1 or np.max(day) > 366:
+        raise ValueError('day must be between 1 and 366')
+    if np.min(hour) < -13. or np.max(hour) > 36.:
+        raise ValueError('hour must be between -13 and 36')
+    if np.min(lat) < -90. or np.max(lat) > 90.:
+        raise ValueError('lat must be between -90 and 90')
+    if np.min(lon) < -180. or np.max(lon) > 180.:
+        raise ValueError('lon must be between -180 and 180')
+
+    # Check array dimensions - need to add check for type (has to be ndarray,
+    # not float. Indexing breaks otherwise)
+    if lat.shape != lon.shape:
+        raise ValueError('Latitude and Longitude must be arrays of same size '
+                         '-- use numpy.meshgrid')
+    if year.shape != day.shape or year.shape != hour.shape:
+        raise ValueError('Time variables must be arrays of same size')
+    if year.shape[0] > 1 and lat.size > 1:
+        if lat.shape != year.shape:
+            raise ValueError('Broadcasting arrays in both time and space '
+                             'requires output from numpy.meshgrid -- all '
+                             'arrays must be the same size')
+
+    # Julian date/Coordinates
+    rpd = np.pi / 180.
+    twopi = np.pi * 2
+    # Add 2,400,000 for true jd
+    # leap = leap days since 1949
+    # 32916.5 is midnite 0 jan 1949 minus 2.4e6
+    delta = year - 1949
+    leap = delta / 4
+    jd = 32916.5 + (delta * 365 + leap + day) + hour / 24.
```

Comment:
I wonder if it would be better to use an exiting package to caluclate the julian date.  `netCDF4.netcdftime` has some nice tools for this.